--- a/cache.h
+++ b/cache.h
@@ -1642,4 +1642,6 @@ void stat_validity_update(struct stat_va
 
 int versioncmp(const char *s1, const char *s2);
 
+extern struct passwd dummy_passwd;
+
 #endif /* CACHE_H */
--- a/wrapper.c
+++ b/wrapper.c
@@ -3,6 +3,16 @@
  */
 #include "cache.h"
 
+struct passwd dummy_passwd = {
+	"dummy", /* user name */
+	"dummy", /* passwd */
+	0, /* user id */
+	0, /* group id */
+	"Dummy User", /* real name */
+	"/", /* home directory */
+	"/bin/bash" /* shell program */
+};
+
 static void do_nothing(size_t size)
 {
 }
@@ -551,9 +561,13 @@ struct passwd *xgetpwuid_self(void)
 
 	errno = 0;
 	pw = getpwuid(getuid());
-	if (!pw)
-		die(_("unable to look up current user in the passwd file: %s"),
-		    errno ? strerror(errno) : _("no such user"));
+	if (!pw) {
+		warning(_("unable to look up current user in the passwd file: %s. Will use dummy."),
+			errno ? strerror(errno) : _("no such user"));
+		dummy_passwd.pw_uid = getuid();
+		dummy_passwd.pw_gid = getgid();
+		return &dummy_passwd;
+	}
 	return pw;
 }
 
