From b99e6df2a6fccadabac90115c447739ae46a1831 Mon Sep 17 00:00:00 2001
From: Kambiz Darabi <darabi@m-creations.net>
Date: Tue, 19 Jan 2016 00:34:06 +0100
Subject: [PATCH 04/10] Do not exit at end of haproxy_main

---
 src/haproxy.c | 2 --
 src/main.c    | 2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -1881,8 +1881,6 @@ int haproxy_main(int argc, char **argv)
 
 	/* Do some cleanup */ 
 	deinit();
-    
-	exit(0);
 }
 
 
--- a/src/main.c
+++ b/src/main.c
@@ -30,6 +30,8 @@
 int main(int argc, char **argv)
 {
 	haproxy_main(argc, argv);
+
+	exit(0);
 }
 
 
