From 40b8e09cc56a2736c3dd51188e6d1a8491736977 Mon Sep 17 00:00:00 2001
From: Kambiz Darabi <darabi@m-creations.net>
Date: Tue, 19 Jan 2016 00:37:44 +0100
Subject: [PATCH 05/10] Makefile: overwrite existing symbolic link for shared
 lib

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/Makefile
+++ b/Makefile
@@ -863,7 +863,7 @@ install-bin:
 	install -d "$(DESTDIR)$(SBINDIR)"
 	install -d "$(DESTDIR)$(LIBDIR)"
 	install libhaproxy.so.$(VERSION) "$(DESTDIR)$(LIBDIR)"
-	ln -rs "$(DESTDIR)$(LIBDIR)/libhaproxy.so.$(VERSION)" "$(DESTDIR)$(LIBDIR)/libhaproxy.so.1"
+	ln -frs "$(DESTDIR)$(LIBDIR)/libhaproxy.so.$(VERSION)" "$(DESTDIR)$(LIBDIR)/libhaproxy.so.1"
 	install haproxy $(EXTRA) "$(DESTDIR)$(SBINDIR)"
 
 install: install-bin install-man install-doc
