From d7e4ac5c8b116cc3461f03b18a8c55eebdf6927e Mon Sep 17 00:00:00 2001
From: Kambiz Darabi <darabi@m-creations.net>
Date: Sat, 23 Jan 2016 16:59:17 +0100
Subject: [PATCH 06/10] Create libhaproxy.so symbolic link

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 42a16e5..d8b6ef7 100644
--- a/Makefile
+++ b/Makefile
@@ -816,6 +816,7 @@ install-bin: haproxy $(EXTRA)
 	install -d "$(DESTDIR)$(LIBDIR)"
 	install libhaproxy.so.$(VERSION) "$(DESTDIR)$(LIBDIR)"
 	ln -frs "$(DESTDIR)$(LIBDIR)/libhaproxy.so.$(VERSION)" "$(DESTDIR)$(LIBDIR)/libhaproxy.so.1"
+	ln -frs "$(DESTDIR)$(LIBDIR)/libhaproxy.so.$(VERSION)" "$(DESTDIR)$(LIBDIR)/libhaproxy.so"
 	install haproxy $(EXTRA) "$(DESTDIR)$(SBINDIR)"
 
 install: install-bin install-man install-doc
-- 
1.9.1

