From 2153b72fdd4295344747f5dca1cfc0d9f10728cf Mon Sep 17 00:00:00 2001
From: Kambiz Darabi <darabi@m-creations.net>
Date: Sat, 23 Jan 2016 17:11:59 +0100
Subject: [PATCH 07/10] Do not call exit in haproxy.c

---
 src/haproxy.c | 68 ++++++++++++++++++++++++++++++-----------------------------
 src/main.c    |  4 +---
 2 files changed, 36 insertions(+), 36 deletions(-)

--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -383,7 +383,6 @@ void usage(char *name)
 		"        -sf/-st [pid ]* finishes/terminates old pids.\n"
 		"\n",
 		name, DEFAULT_MAXCONN, cfg_maxpconn);
-	exit(1);
 }
 
 
@@ -484,7 +483,7 @@ void dump(struct sig_handler *sh)
  * It add only files with .cfg extension.
  * It doesn't add files with name starting with '.'
  */
-void cfgfiles_expand_directories(void)
+int cfgfiles_expand_directories(void)
 {
 	struct wordlist *wl, *wlb;
 	char *err = NULL;
@@ -499,7 +498,7 @@ void cfgfiles_expand_directories(void)
 			Alert("Cannot open configuration file/directory %s : %s\n",
 			      wl->s,
 			      strerror(errno));
-			exit(1);
+			return 1;
 		}
 
 		if (!S_ISDIR(file_stat.st_mode))
@@ -512,7 +511,7 @@ void cfgfiles_expand_directories(void)
 			Alert("Cannot open configuration directory %s : %s\n",
 			      wl->s,
 			      strerror(errno));
-			exit(1);
+			return 1;
 		}
 
 		/* for each element in the directory wl->s */
@@ -531,14 +530,14 @@ void cfgfiles_expand_directories(void)
 			if (!memprintf(&filename, "%s/%s", wl->s, dir_entry->d_name)) {
 				Alert("Cannot load configuration files %s : out of memory.\n",
 				      filename);
-				exit(1);
+				return 1;
 			}
 
 			if (stat(filename, &file_stat)) {
 				Alert("Cannot open configuration file %s : %s\n",
 				      wl->s,
 				      strerror(errno));
-				exit(1);
+				return 1;
 			}
 
 			/* don't add anything else than regular file in cfg_cfgfiles
@@ -551,7 +550,7 @@ void cfgfiles_expand_directories(void)
 				Alert("Cannot load configuration files %s : %s\n",
 				      filename,
 				      err);
-				exit(1);
+				return 1;
 			}
 
 next_dir_entry:
@@ -571,10 +570,10 @@ next_dir_entry:
 }
 
 /*
- * This function initializes all the necessary variables. It only returns
- * if everything is OK. If something fails, it exits.
+ * This function initializes all the necessary variables. It returns
+ * 0 if everything is OK. If something fails, it returns a non-zero error code.
  */
-void init(int argc, char **argv)
+int init(int argc, char **argv)
 {
 	int arg_mode = 0;	/* MODE_DEBUG, ... */
 	char *tmp;
@@ -619,7 +618,7 @@ void init(int argc, char **argv)
 	init_log();
 	signal_init();
 	if (init_acl() != 0)
-		exit(1);
+		return 1;
 	init_task();
 	init_stream();
 	init_session();
@@ -674,7 +673,7 @@ void init(int argc, char **argv)
 				display_version();
 				if (flag[1] == 'v')  /* -vv */
 					display_build_opts();
-				exit(0);
+				return 0;
 			}
 #if defined(ENABLE_EPOLL)
 			else if (*flag == 'd' && flag[1] == 'e')
@@ -733,7 +732,7 @@ void init(int argc, char **argv)
 					oldpids = realloc(oldpids, (nb_oldpids + 1) * sizeof(int));
 					if (!oldpids) {
 						Alert("Cannot allocate old pid : out of memory.\n");
-						exit(1);
+						return 1;
 					}
 					argc--; argv++;
 					oldpids[nb_oldpids] = atol(*argv);
@@ -750,7 +749,7 @@ void init(int argc, char **argv)
 						Alert("Cannot load configuration file/directory %s : %s\n",
 						      *argv,
 						      err_msg);
-						exit(1);
+						return 1;
 					}
 					argv++; argc--;
 				}
@@ -772,7 +771,7 @@ void init(int argc, char **argv)
 						Alert("Cannot load configuration file/directory %s : %s\n",
 						      *argv,
 						      err_msg);
-						exit(1);
+						return 1;
 					}
 					break;
 				case 'p' : cfg_pidfile = *argv; break;
@@ -791,11 +790,12 @@ void init(int argc, char **argv)
 
 	if (change_dir && chdir(change_dir) < 0) {
 		Alert("Could not change to directory %s : %s\n", change_dir, strerror(errno));
-		exit(1);
+		return 1;
 	}
 
 	/* handle cfgfiles that are actualy directories */
-	cfgfiles_expand_directories();
+	if(cfgfiles_expand_directories())
+                return 1;
 
 	if (LIST_ISEMPTY(&cfg_cfgfiles))
 		usage(progname);
@@ -811,13 +811,13 @@ void init(int argc, char **argv)
 		if (ret == -1) {
 			Alert("Could not open configuration file %s : %s\n",
 			      wl->s, strerror(errno));
-			exit(1);
+			return 1;
 		}
 		if (ret & (ERR_ABORT|ERR_FATAL))
 			Alert("Error(s) found in configuration file : %s\n", wl->s);
 		err_code |= ret;
 		if (err_code & ERR_ABORT)
-			exit(1);
+			return 1;
 	}
 
 	pattern_finalize_config();
@@ -828,7 +828,7 @@ void init(int argc, char **argv)
 	err_code |= check_config_validity();
 	if (err_code & (ERR_ABORT|ERR_FATAL)) {
 		Alert("Fatal errors found in configuration.\n");
-		exit(1);
+		return 1;
 	}
 
 	/* recompute the amount of per-process memory depending on nbproc and
@@ -851,7 +851,7 @@ void init(int argc, char **argv)
         err_code |= netns_init();
         if (err_code & (ERR_ABORT|ERR_FATAL)) {
                 Alert("Failed to initialize namespace support.\n");
-                exit(1);
+                return 1;
         }
 #endif
 
@@ -865,7 +865,7 @@ void init(int argc, char **argv)
 	err_code |= srv_init_addr();
 	if (err_code & (ERR_ABORT|ERR_FATAL)) {
 		Alert("Failed to initialize server(s) addr.\n");
-		exit(1);
+		return 1;
 	}
 
 	if (global.mode & MODE_CHECK) {
@@ -883,16 +883,16 @@ void init(int argc, char **argv)
 		if (pr || px) {
 			/* At least one peer or one listener has been found */
 			qfprintf(stdout, "Configuration file is valid\n");
-			exit(0);
+			return 0;
 		}
 		qfprintf(stdout, "Configuration file has no error but will not start (no listener) => exit(2).\n");
-		exit(2);
+		return 2;
 	}
 
 	global_listener_queue_task = task_new();
 	if (!global_listener_queue_task) {
 		Alert("Out of memory when initializing global task\n");
-		exit(1);
+		return 1;
 	}
 	/* very simple initialization, users will queue the task if needed */
 	global_listener_queue_task->context = NULL; /* not even a context! */
@@ -916,12 +916,12 @@ void init(int argc, char **argv)
 		if (err_code & (ERR_ABORT|ERR_FATAL)) {
 			Alert("Failed to initialize filters for proxy '%s'.\n",
 			      px->id);
-			exit(1);
+			return 1;
 		}
 	}
 
 	if (start_checks() < 0)
-		exit(1);
+		return 1;
 
 	if (cfg_maxconn > 0)
 		global.maxconn = cfg_maxconn;
@@ -1019,7 +1019,7 @@ void init(int argc, char **argv)
 			      global.rlimit_memmax,
 			      (int)(mem / (STREAM_MAX_COST + 2 * global.tune.bufsize)),
 			      global.maxconn);
-			exit(1);
+			return 1;
 		}
 
 		if (global.maxsslconn > sides * global.maxconn)
@@ -1059,7 +1059,7 @@ void init(int argc, char **argv)
 			      global.rlimit_memmax,
 			      (int)(mem / (global.ssl_session_max_cost + global.ssl_handshake_max_cost)),
 			      global.maxsslconn);
-			exit(1);
+			return 1;
 		}
 
 		if (global.mode & (MODE_VERBOSE|MODE_DEBUG)) {
@@ -1192,7 +1192,7 @@ void init(int argc, char **argv)
 		      "  %d file descriptors. You should thus reduce global.maxconn by %d. Also,\n"
 		      "  check build settings using 'haproxy -vv'.\n\n",
 		      FD_SETSIZE, global.maxconn, global.maxsock, (global.maxsock + 1 - FD_SETSIZE) / 2);
-		exit(1);
+		return 1;
 	}
 	if (global.mode & (MODE_VERBOSE|MODE_DEBUG)) {
 		printf("Using %s() as the polling mechanism.\n", cur_poller.name);
@@ -1202,11 +1202,11 @@ void init(int argc, char **argv)
 		global.node = strdup(hostname);
 
 	if (!hlua_post_init())
-		exit(1);
+		return 1;
 
 	/* initialize structures for name resolution */
 	if (!dns_init_resolvers(0))
-		exit(1);
+		return 1;
 
 	free(err_msg);
 }
@@ -1676,7 +1676,10 @@ int haproxy_main(int argc, char **argv)
 	char errmsg[100];
 	int pidfd = -1;
 
-	init(argc, argv);
+	err = init(argc, argv);
+	if(err) {
+		return err;
+	}
 	signal_register_fct(SIGQUIT, dump, SIGQUIT);
 	signal_register_fct(SIGUSR1, sig_soft_stop, SIGUSR1);
 	signal_register_fct(SIGHUP, sig_dump_state, SIGHUP);
@@ -1758,14 +1761,14 @@ int haproxy_main(int argc, char **argv)
 			protocol_unbind_all(); /* cleanup everything we can */
 			tell_old_pids(SIGTTIN);
 		}
-		exit(1);
+		return 1;
 	}
 
 	if (listeners == 0) {
 		Alert("[%s.main()] No enabled listener found (check for 'bind' directives) ! Exiting.\n", argv[0]);
 		/* Note: we don't have to send anything to the old pids because we
 		 * never stopped them. */
-		exit(1);
+		return 1;
 	}
 
 	err = protocol_bind_all(errmsg, sizeof(errmsg));
@@ -1777,7 +1780,7 @@ int haproxy_main(int argc, char **argv)
 		protocol_unbind_all(); /* cleanup everything we can */
 		if (nb_oldpids)
 			tell_old_pids(SIGTTIN);
-		exit(1);
+		return 1;
 	} else if (err & ERR_WARN) {
 		Alert("[%s.main()] %s.\n", argv[0], errmsg);
 	}
@@ -1803,7 +1806,7 @@ int haproxy_main(int argc, char **argv)
 			if (nb_oldpids)
 				tell_old_pids(SIGTTIN);
 			protocol_unbind_all();
-			exit(1);
+			return 1;
 		}
 	}
 
@@ -1811,7 +1814,7 @@ int haproxy_main(int argc, char **argv)
 		Alert("[%s.main()] Some configuration options require full privileges, so global.uid cannot be changed.\n"
 		      "", argv[0]);
 		protocol_unbind_all();
-		exit(1);
+		return 1;
 	}
 
 	/* If the user is not root, we'll still let him try the configuration
@@ -1829,7 +1832,7 @@ int haproxy_main(int argc, char **argv)
 			if (nb_oldpids)
 				tell_old_pids(SIGTTIN);
 			protocol_unbind_all();
-			exit(1);
+			return 1;
 		}
 	}
 
@@ -1849,14 +1852,14 @@ int haproxy_main(int argc, char **argv)
 		if (setgid(global.gid) == -1) {
 			Alert("[%s.main()] Cannot set gid %d.\n", argv[0], global.gid);
 			protocol_unbind_all();
-			exit(1);
+			return 1;
 		}
 	}
 
 	if (global.uid && setuid(global.uid) == -1) {
 		Alert("[%s.main()] Cannot set uid %d.\n", argv[0], global.uid);
 		protocol_unbind_all();
-		exit(1);
+		return 1;
 	}
 
 	/* check ulimits */
@@ -1881,7 +1884,7 @@ int haproxy_main(int argc, char **argv)
 			if (ret < 0) {
 				Alert("[%s.main()] Cannot fork.\n", argv[0]);
 				protocol_unbind_all();
-				exit(1); /* there has been an error */
+				return 1; /* there has been an error */
 			}
 			else if (ret == 0) /* child breaks here */
 				break;
@@ -1937,7 +1940,7 @@ int haproxy_main(int argc, char **argv)
 				for (proc = 0; proc < global.nbproc; proc++)
 					while (waitpid(-1, NULL, 0) == -1 && errno == EINTR);
 			}
-			exit(0); /* parent must leave */
+			return 0; /* parent must leave */
 		}
 
 		/* we might have to unbind some proxies from some processes */
@@ -1989,7 +1992,7 @@ int haproxy_main(int argc, char **argv)
 
 	/* initialize structures for name resolution */
 	if (!dns_init_resolvers(1))
-		exit(1);
+		return 1;
 
 	protocol_enable_all();
 	/*
--- a/src/main.c
+++ b/src/main.c
@@ -29,9 +29,7 @@
 
 int main(int argc, char **argv)
 {
-	haproxy_main(argc, argv);
-
-	exit(0);
+	exit(haproxy_main(argc, argv));
 }
 
 
