From 10914bdd9c8fc2b5f06d2ca835a177d91c3f1abc Mon Sep 17 00:00:00 2001
From: Kambiz Darabi <darabi@m-creations.net>
Date: Tue, 26 Jan 2016 23:06:46 +0100
Subject: [PATCH 09/10] Add cffi_rules to act_rules struct

---
 include/types/action.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/types/action.h b/include/types/action.h
index b97f9bf..433be96 100644
--- a/include/types/action.h
+++ b/include/types/action.h
@@ -140,6 +140,7 @@ struct act_rule {
 			int idx;
 		} capid;
 		struct hlua_rule *hlua_rule;
+		struct cffi_rule *cffi_rule;
 		struct {
 			struct sample_expr *expr;
 			const char *name;
-- 
1.9.1

