From 7917e2cb1a7e09b5dbc085c354f43a571dbebc1a Mon Sep 17 00:00:00 2001
From: Kambiz Darabi <darabi@m-creations.net>
Date: Mon, 14 Mar 2016 22:58:15 +0100
Subject: [PATCH 11/11] Just exit at the end of haproxy_main

---
 src/haproxy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/src/haproxy.c
+++ b/src/haproxy.c
@@ -2006,7 +2006,7 @@ int haproxy_main(int argc, char **argv)
 	/* Do some cleanup */
 	deinit();
 
-	return 0;
+	exit(0);
 }
 
 
