--- a/configure.in
+++ b/configure.in
@@ -601,6 +601,8 @@ AC_CACHE_CHECK([for IPv6 support], ac_cv
 #include <netinet/in.h>], [struct sockaddr_in6 s; struct in6_addr t=in6addr_any; int i=AF_INET6; s; t.s6_addr[0] = 0;],
   [ac_cv_ipv6_support=yes], [ac_cv_ipv6_support=no])])
 
+AC_DEFINE([HAVE_FLOCK_LINUX],[1],[Setting flock_type to linux to avoid cross-compiling during configure])
+
 dnl Checks for library functions.
 dnl -------------------------------------------------------------------------
 
