--- a/deps/v8/tools/js2c.py
+++ b/deps/v8/tools/js2c.py
@@ -34,7 +34,6 @@
 import os, re, sys, string
 import optparse
 import jsmin
-import bz2
 import textwrap
 
 
@@ -489,12 +488,7 @@ def CompressMaybe(sources, compression_t
     A sequence of bytes.
   """
   sources_bytes = "".join(sources.modules)
-  if compression_type == "off":
-    return sources_bytes
-  elif compression_type == "bz2":
-    return bz2.compress(sources_bytes)
-  else:
-    raise Error("Unknown compression type %s." % compression_type)
+  return sources_bytes
 
 
 def PutInt(blob_file, value):
@@ -567,7 +561,7 @@ def main():
   parser.set_usage("""js2c out.cc type compression sources.js ...
       out.cc: C code to be generated.
       type: type parameter for NativesCollection template.
-      compression: type of compression used. [off|bz2]
+      compression: type of compression used. [off]
       sources.js: JS internal sources or macros.py.""")
   (options, args) = parser.parse_args()
 
