--- a/Makefile.PL
+++ b/Makefile.PL
@@ -21,7 +21,7 @@ resources
 check_nmake;	# check and download nmake.exe for Win32
 
 requires	'MIME::Base64'	=> 0;	# core module since Perl 5.7.3
-requires	'Convert::ASN1' => 0.20;
+requires	'Convert::ASN1' => 0.27;
 
 test_requires	'Test::More';
 test_requires	'File::Basename';
@@ -37,7 +37,7 @@ features
         ],
 	'LDAPS support' => [
 		-default => 0,
-		'IO::Socket::SSL' => 1.26,
+		'IO::Socket::SSL' => 2.024,
 	],
 	'SASL authentication' => [
 		-default => 0,
